### DISCUSSION TITLE
Feature Request: read json files from s3

### DISCUSSION THREAD

**Discussion by dberardo-com** [duckdb/duckdb#5558] [Ideas]
currently i can make Duckdb read in parquet and csv files from a S3 bucket, but it seems that .json files are completely ignored.

1. how to read in json tables?
2. can one use HIVE style partitions on json files as  well

**Comment by lnkuiper** [comment]
We are planning to improve our json support, but it's a long-time goal. We will support this at some point

**Reply by dberardo-com** [reply]
i see, and in the meantime is it possible to read in JSON objects from in memory Javascript variables in the NodeJS runtime, i mean something like this:

```js
let obj = {a: 1, b: 2}
db.register(obj)
```

currently the only way to achieve something like that would be to first stringify the JSON in javvascript and then create a table from that JSON string.

**Comment by ondrejmo** [comment]
Is there any progress currently planned on this issue? If I understand correctly the support for `hive_partitioning` in csv reader was added by the (not yet released) PR https://github.com/duckdb/duckdb/pull/6912. I would make my usecase significantly easier if I could filter the time range via the pushdown filters, instead of a python wrapper, which I have to use know. Thank you

**Reply by lnkuiper** [reply]
You should be able to read JSON from S3 already, but we do not support hive-partitioned JSON reads yet. I plan to implement the features from #6912 into the JSON reader before the next release.

**Reply by meghfossa** [reply]
Hi @lnkuiper - thank you on this. This would be a game changer for lots of my workflows. Is this slotted for 0.8.x release?

**Reply by lnkuiper** [reply]
Hive-partitioned JSON reads are part of the 0.8.0 release (see: [documentation](https://duckdb.org/docs/extensions/json.html#json-table-functions), it follows the same syntax as for the Parquet/CSV reader). However, we do not support hive-partitioned JSON writes yet.
